### PR TITLE
Shortcircuit paging

### PR DESCRIPTION
**Pull request by ghelloumi**

### CODE REVIEW

File: src/components/results/shortcircuit/shortcircuit-analysis-result.tsx [hunk 1]
```diff
@@ -9,67 +9,166 @@ import WaitingLoader from 'components/utils/waiting-loader';
...
-        </WaitingLoader>
+        <>
+            <WaitingLoader message={'LoadingRemoteData'} loading={isLoading}>
+                <ShortCircuitAnalysisResult
+                    result={result}
+                    onSortChanged={onSortChanged}
+                    sortConfig={sortConfig}
+                />
+            </WaitingLoader>
+            <CustomTablePagination
```
Comment (etiennehomer) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.tsx:164]: There's a scrollbar on the right of this element. There's something to fix on CSS or else

File: src/components/results/shortcircuit/shortcircuit-analysis-result.tsx [hunk 2]
```diff
@@ -9,67 +9,166 @@ import WaitingLoader from 'components/utils/waiting-loader';
...
+            fetchAnalysisResult(
+                fetchShortCircuitAnalysisResult.bind(null, {
+                    studyUuid,
+                    currentNodeUuid: currentNode?.id,
+                    selector,
+                })
+            );
+        } else {
+            fetchAnalysisResult(
+                fetchOneBusShortCircuitAnalysisResult.bind(null, {
```
Comment (etiennehomer) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.tsx:138]: The definition or the call of this method is incorrect

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Comment by etiennehomer** [comment]
The sorting on the busId with Pageable doesn't sort the same way as the front sorting of AgGrid. We'll have to decide what sorting is relevant

**Review by etiennehomer — CHANGES_REQUESTED** [review]
Sorting on Isc min and Isc max does not work

File: src/translations/fr.json [hunk 3]
```diff
@@ -287,6 +287,7 @@
     "ShortCircuitAnalysisParameters": "Paramètres de l'analyse de court-circuit",
     "startShortCircuitError": "Une erreur est survenue lors du lancement de l'analyse de court-circuit",
     "ShortCircuitAnalysisError": "Une erreur est survenue lors de l'exécution de l'analyse de court-circuit",
+    "ShortCircuitAnalysisResultsError": "An error occured when executing short circuit analysis",
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/translations/fr.json]: done, thank you

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 4]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
-export interface ShortcircuitAnalysisResult {
-    resultUuid: UUID;
-    writeTimeStamp: Date;
-    faults: SCAResultFault[];
-}
+type Pageable = {
+    sort: {
+        sorted: boolean;
+        empty: boolean;
+        unsorted: boolean;
```
Comment (Tristan-WorkGH) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:45]: why having `sorted` and `unsorted` at same time? they're always of opposed value no?

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 5]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
+    };
+    pageNumber: number;
+    pageSize: number;
+    offset: number;
+    paged: boolean;
+    unpaged: boolean;
+};
```
Comment (Tristan-WorkGH) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:52]: you can clearly defined that some fields are available only when `unpaged` and `unsorted` are at true with [conditionnal types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) or a combination
```typescript
type Unpaged = {
   readonly unpaged: boolean = true;
}
type Paged = {
    readonly unpaged: boolean = false;
    pageNumber: number;
    pageSize: number;
    offset: number;
}
type Pageable = (Unpaged | Paged) & {
    sort: {
        sorted: boolean;
        empty: boolean;
    };
}
```
or even use nullable sort as indication: `sort?: { empty: boolean },`

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 6]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
+    last: boolean;
+    totalPages: number;
+    totalElements: number;
+    first: boolean;
+    size: number;
+    number: number;
+    sort: {
+        sorted: boolean;
+        empty: boolean;
+        unsorted: boolean;
```
Comment (Tristan-WorkGH) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:66]: why having `sorted` and `unsorted` at same time? they're always of opposed value no?

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 7]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
+    };
+    numberOfElements: number;
+    empty: boolean;
```
Comment (Tristan-WorkGH) [inline src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:69]: isn't a repetition with `size=0`?

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]
i don't know all rules to check

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts (hunk 5 above)
Comment (Tristan-WorkGH) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:52]: it's coming from [Spring Data Pageable](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Pageable.html), so nothing to do...

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts (hunk 6 above)
Comment (Tristan-WorkGH) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:66]: it's coming from [Spring Data Pageable](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/Pageable.html), so nothing to do...

**Review by etiennehomer — CHANGES_REQUESTED** [review]
- the pagination bar is visible on the 'One bus' tab while the pagination is not plugged on this tab
- if you're on a node without result, then you run the short circuit analysis and open the result tab : the shortcircuit results are not displayed

File: src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx [hunk 8]
```diff
@@ -63,72 +67,150 @@ interface ShortCircuitAnalysisResultsFeederResult {
...
-                hide: true,
-            },
-        ];
-    }, [intl]);
+                isHidden: true,
+                isSortable: false,
+            })
+        );
+
+        return columnsDefs;
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx:211]: done, thank you

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 9]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
+    sort: {
+        sorted: boolean;
+        empty: boolean;
+        unsorted: boolean;
+    };
+    pageNumber: number;
+    pageSize: number;
+    offset: number;
+    paged: boolean;
+    unpaged: boolean;
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:51]: I think that we will not use it .. so I remove it, thank you

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts (hunk 4 above)
Comment (ghelloumi) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:45]: I think that we will not use it so I delete it, thank you

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts [hunk 10]
```diff
@@ -40,11 +38,46 @@ export interface SCAResultFault {
...
+    totalElements: number;
+    first: boolean;
+    size: number;
+    number: number;
+    sort: {
+        sorted: boolean;
+        empty: boolean;
+        unsorted: boolean;
+    };
+    numberOfElements: number;
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:68]: I don't have a clear Idea but it is the definition coming from spring data peagable..

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts (hunk 7 above)
Comment (Tristan-WorkGH) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:69]: as @ghelloumi sa, it's server side so nothing to do

File: src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (Tristan-WorkGH) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.type.ts:68]: as @ghelloumi sa, it's server side so nothing to do

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**